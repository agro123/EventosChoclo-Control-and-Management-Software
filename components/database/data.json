{
   "eventos": [
        {
            "id": 123,
            "titulo": "TomorrowLand",
            "fecha_inicial": "27/8/2021",
            "fecha_final": "5/9/2021",
            "num_boletas": 1000,
            "descripcion": "Tomorrowland es un festival de y música electrónica de baile celebrado anualmente en la localidad de Boom.",
            "lugar": "Boom, Amberes",
            "anfitrion": "El jefaso",
            "imagen": "https://upload.wikimedia.org/wikipedia/commons/8/8f/Tomorrowland2015.jpg"
        },
        {
            "id": 456,
            "titulo": "Destroy",
            "fecha_inicial": "12/6/2021",
            "fecha_final": "12/6/2021",
            "num_boletas": 100,
            "descripcion": "Lugar de lokisa",
            "lugar": "Cali",
            "anfitrion": "Desconocido",
            "imagen": "https://image.api.playstation.com/cdn/EP0096/CUSA04545_00/hopaVXL7hdFE6lKafM0TibQDlPERggHu.png"
        },
        {
            "id": 789,
            "titulo": "Final LLA",
            "fecha_inicial": "4/4/2021",
            "fecha_final": "4/4/2021",
            "num_boletas": 1000,
            "descripcion": "Conoceremos al campeón de la Apertura 2021. Sin duda, esta ha sido una de las temporadas más reñidas que hemos tenido.",
            "lugar": "Bogota, Movistar Arena",
            "anfitrion": "League of Legends",
            "imagen": "https://images.contentstack.io/v3/assets/bltad9188aa9a70543a/blt8d8cc6e0b763f8b8/606943eb9a1d7d1050bd4d2c/Final_LLA_Apertura_2021_1600x900.jpg?width=1600&height=900"
        },
        {
            "id": 101,
            "titulo": "Copa America 2021",
            "fecha_inicial": "11/6/2021",
            "fecha_final": "10/7/2021",
            "num_boletas": 36000,
            "descripcion": "La Copa América 2021 será la XLVII edición de este torneo.",
            "lugar": "Colombia-Argentina",
            "anfitrion": "League of Legends",
            "imagen": "https://cnnespanol.cnn.com/wp-content/uploads/2021/02/210224155941-copa-america-argentina-colombia-deportes-cnne-full-169.jpg?quality=100&strip=info"
        }
    ]
}