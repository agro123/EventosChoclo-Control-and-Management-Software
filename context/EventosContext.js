import { createContext, useState } from 'react';


export const EventosContext = createContext();

export const EventosProvider = ({ children }) => {
    const [eventos, setEventos] = useState([]);

    const addEventos = (e) => {
        setEventos(e)
    }

    return (
        <EventosContext.Provider value={
            {
                eventos,
                addEventos
            }
        }>
            {children}
        </EventosContext.Provider>
    )
}
