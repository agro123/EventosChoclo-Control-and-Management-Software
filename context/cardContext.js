import { createContext, useState } from 'react';
import EventoCard from '../components/listarEventos/eventoCard';

export const CardContext = createContext();

export const CardProvider = ({ children }) => {
    const [eventos, setEventos] = useState([]);
    const [data, setData] = useState([]);

    const generarEventos = (d = data) => {
        let cards = [];
        d.map((v, i) => {
            cards = [...cards, (
                <EventoCard
                    info={v}
                    key={i}
                />)]
        })
        setEventos(cards);
    }
    const changeData = d => {
        setData(d);
        generarEventos(d);
    }

    return (
        <CardContext.Provider value={
            {
                eventos,
                generarEventos,
                changeData,
            }
        }>
            {children}
        </CardContext.Provider>
    )
}
