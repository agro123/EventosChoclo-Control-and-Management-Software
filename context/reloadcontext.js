import { createContext, useState } from 'react';

export const ReloadContext = createContext();

export const ReloadProvider = ({ children }) => {
    const [reload, setReload] = useState(true);

    const onChange = () => {
        setReload(!reload);
    }


    return (
        <ReloadContext.Provider value={
            {
                reload,
                onChange
            }
        }>
            {children}
        </ReloadContext.Provider>
    )
}
