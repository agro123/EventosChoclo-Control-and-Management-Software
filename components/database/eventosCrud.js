import data from './data.json';
const { eventos } = data;

export const getEventos = () => {
    return new Promise((resolve, reject) => {
        setTimeout(() => {
            if (eventos) {
                resolve(eventos)
            } else {
                reject('No hay eventos')
            }
        }, 1000)
    })
}

export const deleteEvento = id => {
    return new Promise((resolve, reject) => {
        let found = false;
        setTimeout(() => {
            eventos.map((v, i) => {
                if (v.id === id) {
                    found = true;
                    eventos.splice(i, 1);
                }
            })
            found?resolve(true): reject(false);
        }, 1500)
    })
}

export const getEventoById = id => {
    return new Promise((resolve, reject) => {
        setTimeout(() => {
            const evento = eventos.find( evento => evento.id === id );
            evento? resolve(evento) : reject('Not found');
        }, 1000)
    })
}