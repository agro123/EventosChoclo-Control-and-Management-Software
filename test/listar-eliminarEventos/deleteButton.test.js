import React, { useContext, useEffect } from 'react';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import DeleteButton from '../../components/listarEventos/deleteButton';
import { CardProvider, CardContext } from '../../context/cardContext';
import { getEventoById, getEventos } from '../../components/database/eventosCrud'


const TestComponent = () => {
    const { changeData } = useContext(CardContext);
    useEffect(async () => {
        const data = await getEventos();
        if (data) {
            changeData(data);
        }
    }, [])
    return (<><DeleteButton id={123} /></>)
}

const wrapper = mount(
    <CardProvider >
        <TestComponent />
    </CardProvider>
)

describe('Probando DeleteButton', () => {
    const clickDelete = () => wrapper.find('AntdIcon').simulate('click');
    //pos puede ser 0 para cancelar o 1 para confirmar
    const clickConfirm = (pos) => wrapper.find('button').at(pos).simulate('click')

    /* test('Boton confirmar', async () => {
        clickDelete();
        expect(wrapper.find('DeleteButton')).toMatchSnapshot()
        clickConfirm(1);
        expect(wrapper.find('button').at(1)).toMatchSnapshot()
        const found = await getEventoById(123);
        console.log(found)
        clickConfirm(1); 
        const found = await getEventoById(123);
        expect(found).toMatch('Not found')
        wrapper.unmount(); 
    }) */
    /* test('Boton cancelar', async () => {
        expect(wrapper.find('DeleteButton')).toMatchSnapshot()
        clickDelete();
        clickConfirm(0);
        const found = await getEventoById(123);
        expect(found).toMatch({})
    }) */
})