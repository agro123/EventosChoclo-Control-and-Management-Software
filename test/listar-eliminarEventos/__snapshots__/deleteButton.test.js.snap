// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando DeleteButton Boton confirmar 1`] = `
Array [
  <span
    aria-label="delete"
    className="anticon anticon-delete deleteButton ant-popover-open"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onTouchStart={[Function]}
    role="img"
    tabIndex={-1}
  >
    <svg
      aria-hidden="true"
      data-icon="delete"
      fill="currentColor"
      focusable="false"
      height="1em"
      key="svg-delete"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
        key="svg-delete-svg-0"
      />
    </svg>
  </span>,
  <Portal
    containerInfo={
      <div
        style="position: absolute; top: 0px; left: 0px; width: 100%;"
      >
        <div>
          <div
            class="ant-popover ant-popconfirm"
            style="opacity: 0; pointer-events: none;"
          >
            <div
              class="ant-popover-content"
            >
              <div
                class="ant-popover-arrow"
              >
                <span
                  class="ant-popover-arrow-content"
                />
              </div>
              <div
                class="ant-popover-inner"
                role="tooltip"
              >
                <div
                  class="ant-popover-inner-content"
                >
                  <div
                    class="ant-popover-message"
                  >
                    <span
                      aria-label="exclamation-circle"
                      class="anticon anticon-exclamation-circle"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="exclamation-circle"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm-32 232c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V296zm32 440a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
                        />
                      </svg>
                    </span>
                    <div
                      class="ant-popover-message-title"
                    >
                      Esta acción es irreversible
                    </div>
                  </div>
                  <div
                    class="ant-popover-buttons"
                  >
                    <button
                      class="ant-btn ant-btn-sm"
                      type="button"
                    >
                      <span>
                        Cancel
                      </span>
                    </button>
                    <button
                      class="ant-btn ant-btn-primary ant-btn-sm"
                      type="button"
                    >
                      <span>
                        OK
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    }
  >
    <div>
      <div
        className="ant-popover ant-popconfirm"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "opacity": 0,
            "pointerEvents": "none",
            "zIndex": undefined,
          }
        }
      >
        <div
          className="ant-popover-content"
        >
          <div
            className="ant-popover-arrow"
            key="arrow"
          >
            <span
              className="ant-popover-arrow-content"
            />
          </div>
          <div
            className="ant-popover-inner"
            role="tooltip"
          >
            <div
              className="ant-popover-inner-content"
            >
              <div
                className="ant-popover-message"
              >
                <span
                  aria-label="exclamation-circle"
                  className="anticon anticon-exclamation-circle"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="exclamation-circle"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    key="svg-exclamation-circle"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm-32 232c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V296zm32 440a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
                      key="svg-exclamation-circle-svg-0"
                    />
                  </svg>
                </span>
                <div
                  className="ant-popover-message-title"
                >
                  Esta acción es irreversible
                </div>
              </div>
              <div
                className="ant-popover-buttons"
              >
                <button
                  className="ant-btn ant-btn-sm"
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    key=".0"
                  >
                    Cancel
                  </span>
                </button>
                <button
                  className="ant-btn ant-btn-primary ant-btn-sm"
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    key=".0"
                  >
                    OK
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </Portal>,
]
`;

exports[`Probando DeleteButton No debe de existir el dato en la BD 1`] = `
Array [
  <span
    aria-label="delete"
    className="anticon anticon-delete deleteButton ant-popover-open"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onTouchStart={[Function]}
    role="img"
    tabIndex={-1}
  >
    <svg
      aria-hidden="true"
      data-icon="delete"
      fill="currentColor"
      focusable="false"
      height="1em"
      key="svg-delete"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
        key="svg-delete-svg-0"
      />
    </svg>
  </span>,
  <Portal
    containerInfo={
      <div
        style="position: absolute; top: 0px; left: 0px; width: 100%;"
      >
        <div>
          <div
            class="ant-popover ant-popconfirm"
            style="opacity: 0; pointer-events: none;"
          >
            <div
              class="ant-popover-content"
            >
              <div
                class="ant-popover-arrow"
              >
                <span
                  class="ant-popover-arrow-content"
                />
              </div>
              <div
                class="ant-popover-inner"
                role="tooltip"
              >
                <div
                  class="ant-popover-inner-content"
                >
                  <div
                    class="ant-popover-message"
                  >
                    <span
                      aria-label="exclamation-circle"
                      class="anticon anticon-exclamation-circle"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="exclamation-circle"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm-32 232c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V296zm32 440a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
                        />
                      </svg>
                    </span>
                    <div
                      class="ant-popover-message-title"
                    >
                      Esta acción es irreversible
                    </div>
                  </div>
                  <div
                    class="ant-popover-buttons"
                  >
                    <button
                      class="ant-btn ant-btn-sm"
                      type="button"
                    >
                      <span>
                        Cancel
                      </span>
                    </button>
                    <button
                      class="ant-btn ant-btn-primary ant-btn-sm"
                      type="button"
                    >
                      <span>
                        OK
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    }
  >
    <div>
      <div
        className="ant-popover ant-popconfirm"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "opacity": 0,
            "pointerEvents": "none",
            "zIndex": undefined,
          }
        }
      >
        <div
          className="ant-popover-content"
        >
          <div
            className="ant-popover-arrow"
            key="arrow"
          >
            <span
              className="ant-popover-arrow-content"
            />
          </div>
          <div
            className="ant-popover-inner"
            role="tooltip"
          >
            <div
              className="ant-popover-inner-content"
            >
              <div
                className="ant-popover-message"
              >
                <span
                  aria-label="exclamation-circle"
                  className="anticon anticon-exclamation-circle"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="exclamation-circle"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    key="svg-exclamation-circle"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm-32 232c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V296zm32 440a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
                      key="svg-exclamation-circle-svg-0"
                    />
                  </svg>
                </span>
                <div
                  className="ant-popover-message-title"
                >
                  Esta acción es irreversible
                </div>
              </div>
              <div
                className="ant-popover-buttons"
              >
                <button
                  className="ant-btn ant-btn-sm"
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    key=".0"
                  >
                    Cancel
                  </span>
                </button>
                <button
                  className="ant-btn ant-btn-primary ant-btn-sm"
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    key=".0"
                  >
                    OK
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </Portal>,
]
`;
